Kubernetes/windows/debug/TroubleShootContainerNetworking.ps1

```diff
@@ -0,0 +1,848 @@
+                # Subsequent free ranges go from the end of the previous reserved range to the beginning of the current reserved range
+                $freeRanges += ([Convert]::ToUInt32($filteredTcpRangeArray[$i].P1) - [Convert]::ToUInt32($filteredTcpRangeArray[$i - 1].P2) - 1)
+            }
+    
+            # The last free range goes from the end of the last reserved range to $EphemeralPortEnd
+            $freeRanges += ($EphemeralPortEnd - [Convert]::ToUInt32($filteredTcpRangeArray[$filteredTcpRangeArray.length - 1].P2))
+        }
+        
+        # Count the number of available free ranges
+        [uint32]$freeRangesCount = 0
+        ($freeRanges | % { $freeRangesCount += [Math]::Floor($_ / $portRangeSize) } )
+    
+        return $freeRangesCount
+    }
+    
+    PopulateVfpPortInformation([ref] $VfpPortReference) {
+        [VfpPort]$vfpPort = $VfpPortReference.Value
+        $layerNames = @((vfpctrl.exe /port $vfpPort.Identifier  /list-layer | Out-String -Stream | Select-String -Pattern "LAYER :") -replace "LAYER :")
+            
+        foreach ($layer in $layerNames) {
+            $vfpLayer = [VfpLayer]::new()
+            $vfpLayer.Name = $layer.Trim()
+
+            $groupNames = @((vfpctrl.exe /port $vfpPort.Identifier /layer $vfpLayer.Name /list-group | Out-String -Stream | Select-String -Pattern "GROUP :") -replace "GROUP :")
+            $groupTypes = @((vfpctrl.exe /port $vfpPort.Identifier /layer $vfpLayer.Name /list-group | Out-String -Stream | Select-String -Pattern "Type :" -CaseSensitive) -replace "Type :")
+            $groupDirections = @((vfpctrl.exe /port $vfpPort.Identifier /layer $vfpLayer.Name /list-group | Out-String -Stream | Select-String -Pattern "Direction :") -replace "Direction :")
+            for ($group_num = 0; $group_num -lt $groupNames.Length; $group_num++) {
+                $vfpGroup = [VfpGroup]::new()
+                $vfpGroup.Name = $groupNames[$group_num].Trim()
+                $vfpGroup.Type = $groupTypes[$group_num].Trim()
+                $vfpGroup.Direction = $groupDirections[$group_num].Trim()
+                
+                # Add , at beginning to always convert the output to array even for single element
+                $ruleFriendlyNames = @((vfpctrl.exe /port $vfpPort.Identifier /layer $vfpLayer.Name /group $vfpGroup.Name /list-rule | Out-String -Stream | Select-String -Pattern "Friendly name :") -replace "Friendly name :")
+                $ruleTypes = @((vfpctrl.exe /port $vfpPort.Identifier /layer $vfpLayer.Name /group $vfpGroup.Name /list-rule | Out-String -Stream | Select-String -Pattern "Type :" -CaseSensitive) -replace "Type :")
+                
+                for ($rule_num = 0; $rule_num -lt $ruleFriendlyNames.Length; $rule_num++) {
+                    $vfpRule = [VfpRule]::new()
+                    $vfpRule.Name = $ruleFriendlyNames[$rule_num].Trim()
+                    $vfpRule.Type = $ruleTypes[$rule_num].Trim()
+
```

Comment:
The conditions is empty for me in the endpointdata.json. I assume this is not implemented yet and would come in a future PR.

Reply:
yes. There is more stuff to implement in this script